app/controllers/navigation/Navigator.scala

```diff
@@ -366,12 +389,15 @@ object Navigator {
       case Mode.Draft                                       => controllers.declaration.routes.SummaryController.displayPage(Mode.Draft)
       case _ =>
         val specific = request.declarationType match {
-          case STANDARD      => standardItemPage
-          case SUPPLEMENTARY => supplementaryItemPage
-          case SIMPLIFIED    => simplifiedItemPage
-          case OCCASIONAL    => occasionalItemPage
-          case CLEARANCE     => clearanceItemPage
+          case STANDARD      => standardCacheItemDependent.orElse(standardItemPage)
+          case SUPPLEMENTARY => supplementaryCacheItemDependent.orElse(supplementaryItemPage)
+          case SIMPLIFIED    => simplifiedCacheItemDependent.orElse(simplifiedItemPage)
+          case OCCASIONAL    => occasionalCacheItemDependent.orElse(occasionalItemPage)
+          case CLEARANCE     => clearanceCacheItemDependent.orElse(clearanceItemPage)
+        }
+        commonCacheItemDependent.orElse(commonItem).orElse(specific)(page) match {
+          case mapping: ((Mode, String) => Call)                     => mapping(mode, itemId.id)
+          case mapping: ((ExportsDeclaration, Mode, String) => Call) => mapping(request.cacheModel, mode, itemId.id)
```

Comment:
This has been added by Maciej. Is your branch rebased/updated?

Reply:
Not since Friday.  I'll take a look

Reply:
As per Slack discussion - Maciej did non-item pages.  My changes add support for item pages.